src/server/game/Handlers/MovementHandler.cpp

```diff
@@ -322,21 +322,27 @@ void WorldSession::HandleMovementOpcodes(WorldPacket & recvData)
     // pussywizard: typical check for incomming movement packets
     if (!mover || !mover->IsInWorld() || mover->IsDuringRemoveFromWorld() || guid != mover->GetGUID())
     {
-        recvData.rfinish();
+        recvData.rfinish();                     // prevent warnings spam
         return;
     }
 
     MovementInfo movementInfo;
     movementInfo.guid = guid;
     ReadMovementInfo(recvData, &movementInfo);
 
+    recvData.rfinish();                         // prevent warnings spam
+
     if (!movementInfo.pos.IsPositionValid())
     {
         recvData.rfinish();                     // prevent warnings spam
         return;
     }
 
-    recvData.rfinish();                         // prevent warnings spam
+    if (!mover->movespline->Finalized())
+    {
+        recvData.rfinish();                     // prevent warnings spam
+        return;
+    }
```

Comment:
This check is already contained in this PR https://github.com/azerothcore/azerothcore-wotlk/pull/2410